### PR TITLE
expression: push down left/right/char_length

### PR DESCRIPTION
**Pull request by fzhedu**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tics/issues/1943 

Problem Summary: push down left/right without cosidering collation, char_length aims to verification.

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

How it Works:

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Integration test
- Manual test (add detailed scripts or steps below)

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

- push down left/right/char_length to tiflash

### CODE REVIEW

File: expression/expression.go [hunk 1]
```diff
@@ -1005,9 +1005,12 @@ func scalarExprSupportedByFlash(function *ScalarFunction) bool {
 		ast.TimestampDiff, ast.DateFormat, ast.FromUnixTime,
 		ast.JSONLength:
 		return true
-	case ast.Substr, ast.Substring:
+	case ast.Substr, ast.Substring, ast.Left, ast.Right, ast.CharLength:
 		switch function.Function.PbCode() {
 		case
+			tipb.ScalarFuncSig_LeftUTF8,
+			tipb.ScalarFuncSig_RightUTF8,
+			tipb.ScalarFuncSig_CharLengthUTF8,
```
Comment (zanmato1984) [inline expression/expression.go:1013]: Is this `char length` related to what we are doing?

Comment (fzhedu) [reply expression/expression.go:1013]: open it if needed work around or check when bugs occur

**Review by zanmato1984 — APPROVED** [review]
LGTM

**Pull request merged**